lib/ansible/modules/windows/win_disk_management.ps1

```
@@ -0,0 +1,888 @@
+    }
+    if ($CheckLetter) {
+        if ($SetOnline) {
+            try {
+                Set-OperationalStatus -Disk $disk -Deactivate
+            } catch {
+                $OPStatusFailed = $true
+            } finally {
+                if (-not $OPStatusFailed) {
+                    $result.changed = $true
+                }
+            }
+        }
+        if ($SetWriteable) {
+            try {
+                Set-DiskWriteable -Disk $disk -Deactivate
+            } catch {
+                $ROStatusFailed = $true
+            } finally {
+                if (-not $ROStatusFailed) {
+                    $result.changed = $true
+                }
+            }
+        }
+        Fail-Json -obj $result -message "The drive letter $DriveLetter is set on another partition on this target already which is not allowed"
+    }
+}
+# Check file system
+if ($FileSystem -eq "ntfs") {
+    if ($Size -le 256000) {
+    } else {
+        if ($SetOnline) {
+            try {
+                Set-OperationalStatus -Disk $disk -Deactivate
+            } catch {
+                $OPStatusFailed = $true
+            } finally {
+                if (-not $OPStatusFailed) {
+                    $result.changed = $true
+                }
+            }
+        }
+        if ($SetWriteable) {
+            try {
+                Set-DiskWriteable -Disk $disk -Deactivate
+            } catch {
+                $ROStatusFailed = $true
+            } finally {
+                if (-not $ROStatusFailed) {
+                    $result.changed = $true
+                }
+            }
+        }
+        Fail-Json -obj $result -message "Option size with value $($Size)gb is not a valid size for ntfs hence the disk can not be formatted with this file system"
+    }
+} elseif ($FileSystem -eq "refs") {
+    if ($AllocUnitSize -ne 64) {
+        $AllocUnitSize = 64    
```

This should throw a failure as the unit size people specify is invalid with the refs file system.